client.go

```diff
@@ -70,6 +70,7 @@ func (c *Client) Idle(stop <-chan struct{}) error {
 			close(stopOrRestart)
 			return <-done
 		case err := <-done:
+			close(stopOrRestart)
```

Comment:
Hmm, but this should only happen when `c.idle` has returned right?

Reply:
Yeah, sorry for not providing a clear explanation. If `stopOrRestart` isn't closed here, a goroutine started in `Response` continues waiting for `r.Stop` and never ends: https://github.com/emersion/go-imap-idle/blob/2704abd7050ed7f2143753554ee23affdf847bd9/response.go#L30-L33

Reply:
Ah, I see. And it doesn't really make sense to close it in `idle` because we're creating the channel.